src/algoliasearch.helper.js

```diff
@@ -1201,7 +1212,13 @@ AlgoliaSearchHelper.prototype._search = function() {
 
   this._currentNbQueries++;
 
-  this.client.search(queries, this._dispatchAlgoliaResponse.bind(this, states, queryId));
+  try {
+    this.client.search(queries)
+      .then(this._dispatchAlgoliaResponse.bind(this, states, queryId))
+      .catch(this._dispatchAlgoliaError.bind(this, queryId));
+  } catch (err) {
+    this.emit('error', err);
```

Comment:
You could probably add a comment here explaining that if we reach this part, we're in a internal error state.